### DISCUSSION TITLE
Time type still seems to have timezone issue, even after the fix #10956

### DISCUSSION THREAD

**Discussion by Senorsen** [directus/directus#13550] [Archive]
### Describe the Bug

Time type still seems to have timezone issue, even after the fix #10956.

After creating item with the `time` or `datetime` type field empty, which has 'On Create: Save Current Date/Time' setting, the field seems to be set as UTC time, instead of current timezone.

### To Reproduce

For example, I'm in the GMT+8 timezone, and now time is about 12:48 PM.

I created mainly 3 fields:
| Name | Type | Setting |
| --- | --- | --- |
| time | `time` | On Create: Save Current Date/Time; Not allow null value |
| createdAt | `datetime` | On Create: Save Current Date/Time; Not allow null value |
| createdAt2 | `datetime` | Default Value: `NOW()`; Not allow null value |

After creating an item with these fields empty, only the `createdAt2` field is correct.
**Figure 1: creating an item with these fields empty**
<img width="918" alt="screenshot 2022-05-25 12 48 45" src="https://user-images.githubusercontent.com/4176091/170182026-f9056533-b380-4d23-a9a3-7c3a2bbbfed9.png">

**Figure 2: after creating the item**
<img width="924" alt="screenshot 2022-05-25 12 49 45" src="https://user-images.githubusercontent.com/4176091/170182120-b36dc61f-76d9-41fe-a3f5-d765a69b22b2.png">

### What version of Directus are you using?

9.11.1

### What version of Node.js are you using?

Using official docker image, seems to be v16.15.0

### What database are you using?

MySQL 8.0.27 (Source distribution) (using bitnami's chart)

### What browser are you using?

Chrome 101.0.4951.64

### How are you deploying Directus?

Kubernetes 1.22.4, containerd 1.6.4

**Comment by azrikahar** [comment]
Although I was able to replicate the same behavior locally (GMT+8, which is UTC+8):

![chrome_BoM6K6PXkf](https://user-images.githubusercontent.com/42867097/170212852-a7e4036e-09f8-4932-8e8b-825552e540e0.png)

PR #10956 does explicitly state that the _server_ should be running on UTC (+0). By adhering to this principle and running Directus on a VPS with UTC+0, I can see that the results are the same now:

![chrome_1qofRwrmvA](https://user-images.githubusercontent.com/42867097/170213022-d8ea84a9-775f-4ef8-bb6a-7cab37e8abff.png)

So it seems like it's more of a potential improvement to the docs to highlight this caveat post-PR #10956, provided I'm not misunderstanding anything here. cc our time lord @licitdev for confirmation!

**Comment by Senorsen** [comment]
@azrikahar Thanks for your nice explanation! My directus server is indeed running on UTC (+0), but the MySQL server is running on UTC+8, so this might be the problem.

I'm looking for some configuration env key like `DB_TIMEZONE` but seems there's no such configuration key...

**Comment by azrikahar** [comment]
> I'm looking for some configuration env key like DB_TIMEZONE but seems there's no such configuration key...

I believe it does exist for Knex (any `DB_` prefixed env variables will be passed to it), but it maybe dependent on the driver if I'm not mistaken. Did you have any luck with using that flag?

**Comment by licitdev** [comment]
If I recall correctly, `DB_TIMEZONE="+00:00"` can be set for MySQL.
The issue here is the `NOW()`, which becomes `CURRENT_TIMESTAMP` will not work as intended if the database isn't in the UTC timezone. 

I'd recommend to select the `Save Current Date/Time` option under `On Create` for date/time fields.
